Documentation/ExtensionConfiguration/Index.rst

```diff
@@ -47,13 +48,66 @@ extension. The content should look as follows::
 The values can and should be customized of course. A more meaningful and longer
 description is recommended and defining some details about the author
 (`author`, :code:`author_email` and :code:`author_company`) make also perfect sense. A
-detailed description of all configuration options can be found in the
-:ref:`Core API Reference <t3coreapi:extension-declaration>`.
+detailed description of all configuration options can be found in
+:ref:`TYPO3 Explained: Declaration file <t3coreapi:extension-declaration>`.
 
 Create and customize this file and store it as :file:`site_package/ext_emconf.php`.
-The configration shown above will do the job, if you do not want to customize
+The configuration shown above will do the job, if you do not want to customize
 the declaration file at the moment.
 
+Composer configuration
+======================
+
+If you are planning to work with a Composer-based installation (as we would
+advise) the extension needs to contain its own :file:`composer.json`.
+
+.. code-block:: json
+
+   {
+      "name": "t3docs/site-package",
+      "type": "typo3-cms-extension",
+      "description": "Example site package from the site package tutorial",
+      "authors": [
+         {
+            "name": "TYPO3 CMS Documentation Team",
+            "role": "Developer",
+            "homepage": "https://typo3.org/community/teams/documentation"
+         },
+         {
+            "name": "The TYPO3 Community",
+            "role": "Contributor",
+            "homepage": "https://typo3.org/community/"
+         }
+      ],
+      "homepage": "https://github.com/TYPO3-Documentation/TYPO3CMS-Tutorial-SitePackage-Code",
+      "license": "MIT",
+      "keywords": [
+         "typo3",
+         "site package",
+         "documentation"
+      ],
+      "support": {
+         "issues": "https://github.com/TYPO3-Documentation/t3docs-screenshots/issues"
+      },
+      "extra": {
+         "typo3/cms": {
+            "extension-key": "site_package"
+         }
+      }
+   }
+
+For historic reasons TYPO3 extension names have to be written in lower case
+separated by underscores. The path name and the extension name have to be
+the same. So the extension in the path :file:`site_package` has to have the
+same "extension-key" to be defined in the "extra" section of the composer.json.
```

Comment:
There are no requirements any more for local paths with a given "extension-key". I suggest to make it a suggestion